src/Application.java

```diff
@@ -0,0 +1,58 @@
+import sun.java2d.loops.TransformBlit;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+public class Application {
+
+    public void start() {
+        Transaction[] transactions = getData();
+        Transaction[] orderedTransactions = orderTransactions(transactions);
+        for (Transaction transaction : orderedTransactions) {
+            print(transaction);
+        }
+    }
+
+    public Transaction[] getData() {
+        return new TransactionData().getTransactionData();
+    }
+
+    public Transaction[] orderTransactions(Transaction[] unorderedTransactions) {
+        List<Transaction> transactionList = Arrays.asList(unorderedTransactions);
+        transactionList.sort(new Comparator<Transaction>() {
+            @Override
+            public int compare(Transaction o1, Transaction o2) {
+                return o1.getDate().compareTo(o2.getDate());
+            }
+        });
+        return transactionList.toArray(new Transaction[unorderedTransactions.length]);
+    }
+
+    public void print(Transaction transaction) {
```

Comment:
Methods that just print should be suuuuuuuuuuper simple, and preferably have their own designated type (often called `Logger`) to do so.
The reason is for modularity, but is really well shown through tests:
1) I do not recommend attempting to test that something got printed to the screen. It's NOT pretty. However, it **is** useful to test if a Transaction got formatted correctly.
2) If we wrap the `System.out.println()` call in a class of its own, we can make it a dependency (constructor parameter) of e.g. the `Application` class, and mock it out in our test (so that we can verify that it was called with the correct arguments. It might be hard to visualise before seeing it for yourself - so we'll go over this in more detail tomorrow.

So (foreshadowing what's going to happen tomorrow) what I would do is something like this:

```
TransactionFormatter (class)
TransactionFormatterTest (class)
Logger (interface)
DefaultLogger (class)
```

`TransactionFormatter.java`:
```java
public class TransactionFormatter {
    public String format(Transaction transaction) {
        //Turn that transaction into a String, like you're doing inside your `System.out.Println` call right now.
        //Test-drive this with TransactionFormatterTest.java
    }
}
```
`Logger.java`:
```java
public interface Logger {
    void print(String message);
}
```

`DefaultLogger.java`:
```java
public class DefaultLogger implements Logger {
    @Override
    public void print(String message) {
        System.out.println(message);
    }
}
```

Reply:
Redesign this for sprint-4